// Copyright 2023 Intrinsic Innovation LLC

syntax = "proto3";

package intrinsic_proto.skills;

import "google/protobuf/duration.proto";
import "intrinsic/skills/proto/skills.proto";

message PythonSkillServiceConfig {
  // The list of module names that need to be imported to run this skill
  // service.
  repeated string module_names = 1;
}

message ExecutionServiceOptions {
  // The amount of time a skill has to prepare for cancellation.
  google.protobuf.Duration cancellation_ready_timeout = 2;
}

message SkillServiceConfig {
  // The name of the skill. This should match the name given by the skill's
  // implementation.
  string skill_name = 1 [deprecated = true];

  // The binary proto file containing the parameters FileDescriptorSet.
  string parameter_descriptor_filename = 3 [deprecated = true];

  // The binary proto file containing the return values FileDescriptorSet.
  string return_value_descriptor_filename = 4 [deprecated = true];

  // The binary proto file containing the published topics FileDescriptorSet.
  string pub_topic_descriptor_filename = 5 [deprecated = true];

  // Language-specific configuration options.
  oneof language_specific_options {
    PythonSkillServiceConfig python_config = 10;
  }

  Skill skill_description = 6;

  ExecutionServiceOptions execution_service_options = 7;

  reserved 2;
  reserved "image_name";
}
